tests/test_edgeql_expressions.py

```diff
@@ -9905,3 +9905,200 @@ async def test_edgeql_cast_to_function_01(self):
+            create module std::test;
+            create scalar type std::test::Foo extending int64;
+        """)
+
+        valid_queries = [
+            'select <test::Foo>1;',
+            'select <std::test::Foo>1;',
+            'with module dummy select <test::Foo>1;',
+            'with module dummy select <std::test::Foo>1;',
+            'with module test select <Foo>1;',
+            'with module test select <test::Foo>1;',
+            'with module test select <std::test::Foo>1;',
+            'with module std select <test::Foo>1;',
+            'with module std select <std::test::Foo>1;',
+            'with module std::test select <Foo>1;',
+            'with module std::test select <test::Foo>1;',
+            'with module std::test select <std::test::Foo>1;',
+        ]
+        invalid_queries = [
+            'select <Foo>1;',
+            'with module dummy select <Foo>1;',
+            'with module std select <Foo>1;',
+        ]
+
+        for query in valid_queries:
+            await self.con.execute(query)
+
+        for query in invalid_queries:
+            async with self.assertRaisesRegexTx(
+                edgedb.errors.InvalidReferenceError,
+                "Foo' does not exist",
+            ):
+                await self.con.execute(query)
+
+    async def test_edgeql_expr_with_module_04(self):
+        await self.con.execute(f"""
+            create module dummy;
+            create module std::test;
+            create scalar type std::test::Foo extending int64;
+            create module test;
+        """)
+
+        valid_queries = [
+            'select <std::test::Foo>1;',
+            'with module dummy select <std::test::Foo>1;',
+            'with module test select <std::test::Foo>1;',
+            'with module std select <std::test::Foo>1;',
+            'with module std::test select <Foo>1;',
+            'with module std::test select <std::test::Foo>1;',
+        ]
+        invalid_queries = [
+            'select <Foo>1;',
+            'select <test::Foo>1;',
+            'with module dummy select <Foo>1;',
+            'with module dummy select <test::Foo>1;',
+            'with module test select <Foo>1;',
+            'with module test select <test::Foo>1;',
+            'with module std select <Foo>1;',
+            'with module std select <test::Foo>1;',
+            'with module std::test select <test::Foo>1;',
+        ]
+
+        for query in valid_queries:
+            await self.con.execute(query)
+
+        for query in invalid_queries:
+            async with self.assertRaisesRegexTx(
+                edgedb.errors.InvalidReferenceError,
+                "Foo' does not exist",
+            ):
+                await self.con.execute(query)
+
+    async def test_edgeql_expr_with_module_05(self):
+        await self.con.execute(f"""
+            create module dummy;
+            create module std::test;
+            create function std::test::Foo(x: int64) -> int64 using(x);
```

Comment:
There is already a `std::_test` module and a `std::_test::abs` function; we can use those